domino/domino.py

```diff
@@ -423,7 +427,7 @@ def validate_spark_executor_count(executor_count, max_executor_count):
         def get_default_spark_settings():
             self.log.debug("Getting default spark settings")
             default_spark_setting_url = self._routes.default_spark_setting(
-                self._project_id
+                self.project_id
```

Comment:
What are the tradeoffs involved in making this no longer private?

Reply:
In addition to using it internally (as in the example above), it can also be used externally, particularly within datasets to filter current project datasets. It can also be used, can cases where someone may want to switch between projects, in ensuring current project.

Reply:
Keeping it private would mean that users would be force to use a private property to reach the same goal.